projects/js-packages/publicize-components/src/social-store/selectors/shares-data.ts

```diff
@@ -0,0 +1,62 @@
+import { isWpcomPlatformSite } from '@automattic/jetpack-script-data';
+import { store as coreStore } from '@wordpress/core-data';
+import { createRegistrySelector } from '@wordpress/data';
+import { getSocialScriptData } from '../../utils';
+
+/**
+ * Get the shares data from the store or script data
+ * @param select - Select function
+ * @return The shares data
+ */
+const getSharesData = select => {
+	// On WordPress.com platform sites, use script data directly
+	if ( isWpcomPlatformSite() ) {
+		return getSocialScriptData().shares_data || {};
+	}
```

Comment:
Is this definitely necessary? The shares data will be initialised, but if we invalidate the resolution, I suppose we would make an unnecessary API call.

If we could avoid this, it would be great. Having different behaviour in the different environments sometimes causes us problems.

Reply:
Yes, the selector should ideally be unaware of this logic. Whether to show the shares data or not should be handled in the component itself.

Reply:
I don't think it's about whether to show the shares data, but where it gets it from. It may not be necessary though as we have never enabled the share limits on wpcom, so shouldn't be displaying the cards and therefore shouldn't be calling this selector.

We could probably rely on that, as the worse that would happen is an extra API call to update the values.